hnn_core/feed.py

```diff
@@ -180,13 +180,17 @@ def __create_extinput(self):
         """Creates the ongoing external inputs (rhythmic)."""
         # print("__create_extinput")
         # Return if all synaptic weights are 0
+        all_syn_weights_zero = True  # fixes #101
         for key in self.p_ext.keys():
             if key.startswith('L2Pyr') or \
                     key.startswith('L5Pyr') or \
                     key.startswith('L2Bask') or \
                     key.startswith('L5Bask'):
-                if self.p_ext[key][0] <= 0.0:
-                    return False
+                if self.p_ext[key][0] > 0.0:
+                    all_syn_weights_zero = False
+        if all_syn_weights_zero:
+            return False
```

Comment:
it's a bit weird that this method even returns something because the return value is not used anywhere

Reply:
@cjayb I see, so the error in #101 is caused by one of the synaptic weights from 'OnlyRhythmicDist.param' being zero, thus causing `__create_extinput()` to return `False` even though some of the other weights are non-zero?

Reply:
To address @jasmainak's point, this might be a good time to rethink the return values of the `__create_` methods in the `ExtFeed` class. It looks like the code is currently set up to return the number of inputs added to the feed (for troubleshooting purposes?), though this might be unnecessary if we implement good tests.

Reply:
yes, +10 for tests! @rythorpe do you have a suggestion for a test based on your experience in #101 ?

Reply:
> so the error in #101 is caused by one of the synaptic weights from 'OnlyRhythmicDist.param' being zero

Correct, these are the only non-zero weights:

```python
    'input_dist_A_weight_L2Pyr_ampa': 5.4e-5,
    'input_dist_A_weight_L5Pyr_ampa': 5.4e-5,
```

The method would return unless _all_ weights were non-zero.

> rethink the return values of the `__create_` methods in the `ExtFeed` class

+1 It doesn't seem that the return values are ever used

Reply:
> if we implement good tests

Right :) So how do we do this? I can update this PR to remove the return arguments, and replace with a test. But how?

As I mentioned, running the full alpha example takes too long. We could use a smaller `'tstop'`, then do something like in #99 for the `spikegids` of type `'extinput'`?

Reply:
I think all these are great ideas for faster tests. I'm still having trouble understanding what exactly you fixed though. It seems to me that all the logic in the diff is modifying the return value but the return value is not used anywhere? What exactly changed from `master`? Maybe a good next step would be to update `whats_new.rst` so we write a description of the fix in plain english!

Reply:
Follow the loop logic: it should only end the function if _all_ inputs are zero. Before it would end if _any_ input was <=0.
I’ll update the rst, but I think this was an unintentional mod (bug)

Reply:
okayyyy, now I get it

so all you need to do for the unit test is to initialize an `ExtFeed` object with the right parameters and then check if `self.eventvec` exists (or is not None. I didn't check exactly what this function does). No computation required at all.

Reply:
> I’ll update the rst, but I think this was an unintentional mod (bug)

yep it should go to the bug section of `whats_new.rst`

Reply:
Continuing off @jasmainak's suggestion, ideally we'd check the number and type of inputs in the `p_ext` attribute of an `ExtFeed` object so that we can be sure all (vs. any) expected inputs were added to the feed. I think this test could be added to the network test script.

Reply:
see my latest push: it wasn't enough to check `p_ext`, since these could be copied in fine, despite `__create_extinput` not running through to create the `eventvec`. I think we should test `p_ext` in `create_pext`?